src/components/modal/AddPost.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div class="fixed inset-0 flex items-center justify-center z-50">
+    <div
+      class="bg-white w-3/4 h-3/4 md:w-2/3 md:h-2/3 lg:w-3/4 lg:h-3/4 p-6 rounded-lg shadow-lg text-center"
+    >
+      <BaseForm 
+        :actions="false" 
+        @submit="onSubmit">
+          <div class="flex relative items-center justify-center mb-5">
+            <h1 class="font-bold">{{ $t("Add Post") }}</h1>
+
+            <XIcon
+              size="2.5x"
+              class="cursor-pointer absolute right-2"
+              @click="$emit('close-modal')"
+            />
+          </div>
+
+          <div>
+            <FormKit
+              v-model="newPost.title"
+              type="text"
+              label="Title"
+              validation="required|length:0,50"
+            />
+            <FormKit
+              v-model="newPost.body"
+              type="textarea"
+              label="Content"
+              validation="required|length:0,500"
+            />
+            <Tags 
+              v-if="newPost?.tags?.length"
+              :tags="newPost.tags"
+              :isClearable="true"
+              class="justify-center"
+              @remove-tag="removeTag"
+            />
+            <div class="grid justify-center mt-5 mb-20">
```

Comment:
move this div along with addTag and removeTag functions in Tags component, and display it dynamically (by prop 'canAdd')

hint: handle tags with v-model